### DISCUSSION TITLE
When does urql send a `OPTIONS` request?

### DISCUSSION THREAD

**Discussion by alea12** [urql-graphql/urql#1659] [Q&A] [answered]
I'm testing urql along with [GraphQL Ruby](https://graphql-ruby.org/).
When I launch the backend Rails server and make a simple query request like this:

```js
import { createClient, Provider, useQuery } from "urql";

const urqlClient = createClient({ url: "http://localhost:3000/graphql" });

const postQuery = `
  query PostQuery($id: ID!) {
    post(id: $id) {
      id
      title
    }
  }
`;

function PostDetail({ id }) {
  const [result] = useQuery({
    query: postQuery,
    variables: { id },
    requestPolicy: "cache-and-network",
  });
  const { data, error, fetching } = result;
  if (error) return `err: ${error.message}`;
  if (fetching) return "fetching...";

  const post = data.post;

  return (
    <Provider value={urqlClient}>
      <h2>{post.title}</h2>
    </Provider>
  );
}
```

There seems to be a `OPTIONS` request sent to the backend initially (which GraphQL Ruby does not support out of the box), but after that first request, `POST` requests are used when making query / mutation requests. 

When does urql send a `OPTIONS` request? Is it checking the CORS policy or something with it, when the `<Provider />` is loaded for the first time?

**Comment by codingnuclei** [comment, marked as answer]
It is to do with CORS and how the browser automatically handles CORS requests.

Take a look [https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS](MDN)

So basically it isn't anything to do with URQL :)

**Reply by alea12** [reply]
Makes sense, thanks so much!
